### PR TITLE
feat(cart): make cart clear API return the new cart

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Depends on #572

## What is the current behavior?
Currently, when you clear, you're left without a new cart object to work with.

## What is the new behavior?
This returns a new cart object when you clear the cart.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/cart/src/effects/cart.effects.ts [hunk 1]
```diff
@@ -52,4 +54,17 @@ export class DaffCartEffects {
         )
     )
   )
+
+  @Effect()
+  clearCart$ = this.actions$.pipe(
+    ofType(DaffCartActionTypes.CartResetAction),
+    switchMap(() =>
+      this.driver.clear().pipe(
+          map((resp) => new DaffCartResetSuccess(resp)),
```
Comment (Nolan-Arnold) [inline libs/cart/src/effects/cart.effects.ts]: I believe this should be `return new DaffCartResetSuccess(resp)),`

File: libs/cart/src/effects/cart.effects.ts [hunk 2]
```diff
@@ -10,7 +10,9 @@ import {
   DaffCartLoadFailure, 
   DaffAddToCartSuccess,
   DaffAddToCartFailure,
-  DaffAddToCart} from '../actions/cart.actions';
+  DaffAddToCart,
+  DaffCartResetSuccess,
+  DaffCartResetFailure} from '../actions/cart.actions';
 import { DaffCartDriver } from '../drivers/injection-tokens/cart-driver.token';
 import { DaffCartServiceInterface } from '../drivers/interfaces/cart-service.interface';
```
Comment (Nolan-Arnold) [inline libs/cart/src/effects/cart.effects.ts]: `DaffCartEffects` should now be `DaffCartEffects<T extends DaffCart>`

File: libs/cart/src/effects/cart.effects.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/cart/src/effects/cart.effects.ts]: @Nolan-Arnold this is a short-hand equivalent of the same.

```
() => { return "string"; }
() => "string";
function(){
  return "string";
}
```

are all equivalent in `es6`

**Comment by lderrickable** [comment]
Why do we want a new cart instead of just an old, empty cart? To remove the selected shipping address and payment method? I would think we'd want to keep those.

**Comment by damienwebdev** [comment]
> Why do we want a new cart instead of just an old, empty cart? To remove the selected shipping address and payment method? I would think we'd want to keep those.

This implementation will be dependent on each platform driver. This merely updates the interface so that end users don't have to make additional driver calls after clearing, they'll get back SOME new cart, depending on what the platform does behind the scenes.

**Comment by lderrickable** [comment]
ok sure. lgtm

File: libs/cart/src/effects/cart.effects.ts [hunk 3]
```diff
@@ -53,4 +55,17 @@ export class DaffCartEffects<T extends DaffCart> {
         )
     )
   )
+
+  @Effect()
+  clearCart$ = this.actions$.pipe(
```
Comment (lderrickable) [inline libs/cart/src/effects/cart.effects.ts]: This needs tested

**Comment by damienwebdev** [comment]
@lderrickable needs re-review

**Comment by Nolan-Arnold** [comment]
I'm not 100% sure why codeclimate doesn't like this code.

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**